indico/modules/events/abstracts/util.py

```diff
@@ -63,42 +79,82 @@ def _get_static_columns(self, ids):
 
         :return: a list of {'id': ..., 'caption': ...} dicts
         """
-        return [{'id': id_, 'caption': self.static_items[id_]['title']} for id_ in ids if id_ in self.static_items]
+        return [{'id': id_, 'caption': self.static_items[id_]['title']} for id_ in self.static_items if id_ in ids]
+
+    def _get_sorted_contribution_fields(self, item_ids):
+        """Return the contribution fields ordered by their position in the abstract form."""
+
+        return (ContributionField
+                .find(ContributionField.id.in_(item_ids))
+                .with_parent(self.list_event)
+                .order_by(ContributionField.position)
+                .all())
+
+    def _get_filters_from_request(self):
+        filters = super(AbstractListGenerator, self)._get_filters_from_request()
+        for field in self.list_event.contribution_fields:
+            if field.field_type == 'single_choice':
+                options = request.form.getlist('field_{}'.format(field.id))
+                if options:
+                    filters['fields'][str(field.id)] = options
+        return filters
 
     def _build_query(self):
         return (Abstract.query
                 .with_parent(self.list_event)
-                # TODO: .filter(~Abstract.is_deleted)
-                # TODO: add options to `joinedload` fields (once implemented)
+                .filter(~Abstract.is_deleted)
                 .order_by(Abstract.id))
 
     def _filter_list_entries(self, query, filters):
-        if not (filters.get('fields') or filters.get('items')):
-            return query
-        # TODO: Get field filters
-        field_filters = {}
-        if not field_filters and not filters['items']:
-            return query
-        # TODO: Construct field items criteria
-        # items_criteria = []
         criteria = []
-        static_filters = {
-            # TODO: 'session': Contribution.session_id,
-            'final_track': Abstract.final_track_id,
-            # TODO: 'proposed_track': Abstract.track_id,
-            'final_type': Abstract.final_type_id,
-            'proposed_type': Abstract.type_id
-        }
-        for key, column in static_filters.iteritems():
-            ids = set(filters['items'].get(key, ()))
-            if not ids:
-                continue
-            column_criteria = []
-            if None in ids:
-                column_criteria.append(column.is_(None))
-            if ids - {None}:
-                column_criteria.append(column.in_(ids - {None}))
-            criteria.append(db.or_(*column_criteria))
+        field_filters = filters.get('fields')
+        item_filters = filters.get('items')
+        extra_filters = filters.get('extra')
+
+        if not (field_filters or item_filters or extra_filters):
```

Comment:
Well, I read it "if there is no X or Y or Z". Not so sure if it's grammatically correct tho! Perhaps "if there is no X nor Y nor Z".